bin/options.js

```diff
@@ -17,37 +17,54 @@ module.exports = {
 };
 
 /**
- * Default pathname for run-control file.
+ * Default test directory.
  *
  * @constant
  * @type {string}
  * @default
  */
-const defaultPathname = 'test/mocha.opts';
+const DEFAULT_TEST_DIRECTORY = 'test';
+
+/**
+ * Default filename of run-control file.
+ *
+ * @constant
+ * @type {string}
+ * @default
+ */
+const DEFAULT_OPTS_FILENAME = 'mocha.opts';
 
 /**
  * Reads directories.test of package.json.
  *
  * @public
- * @returns {string} test directory, if it exists
+ * @returns {string} test directory
  */
+let cachedTestDirectory;
+
 function getTestDirectory() {
-  if (!getTestDirectory.cache) {
+  if (!cachedTestDirectory) {
+    cachedTestDirectory = DEFAULT_TEST_DIRECTORY;
     try {
-      const pkg = readPkgUp.sync('__dirname').path;
-      const testDir = require(pkg).directories.test;
-      if (testDir) {
-        getTestDirectory.cache = testDir;
-      } else {
-        getTestDirectory.cache = 'test';
+      let pkgPath = readPkgUp.sync(process.cwd()).path;
+
+      if (!pkgPath) {
+        // PIW: get directory of specific test location
+        const specTest = path.dirname(process.argv[process.argv.length - 1]);
+        pkgPath = readPkgUp.sync(specTest).path;
+      }
+
+      if (pkgPath) {
+        const pkg = require(pkgPath);
```

Comment:
Confused with _all_ of the above. Since `readPkgUp.sync` will return the package contents via `.pkg`, why go through all this?

Reply:
@plroebuck in most cases "package.json" is located in `cwd` or one of its parent directories. `npm test` throws an error otherwise. With `node mocha` there is no guaranty that `cwd` is user's project root, please see above in "Additional information". So in a second step I'm trying to use the last argv element, if it is a path with/without glob, to get a second entry point for the search of "package.json". This is more complicated, since argv has not been processed at that point.

It would be far easier to just use `cwd`/parent dirs and throw an error otherwise as `npm`does.
And no, user's project root can't be located in any case.
I would prefer the way of `npm`, no second entry point and throw an error.

Reply:
@plroebuck this is a quote out of the chat room:
> mocha’s behavior is really just “best effort". we can’t know if the root is cwd, but most people are probably running mocha with the cwd of the root, so we can try ./test.

So, Mocha does one search with cwd as a starting point, there is no second search and no error will be thrown. Mocha will behave as in the current release, just the fix default value `./test` can be replaced by `directories.test` of "package.json".